skunkworks_crow/src/main/java/org/odk/share/utilities/DialogUtils.java

```diff
@@ -21,48 +27,80 @@ private DialogUtils() {
+        return alertDialog;
+    }
+
+    /**
+     * Create an {@link AlertDialog} for switching receive method.
+     */
+    public static AlertDialog createMethodSwitchDialog(@NonNull Activity targetActivity, DialogInterface.OnClickListener listener) {
+        AlertDialog alertDialog = new AlertDialog.Builder(targetActivity).create();
+        alertDialog.setCancelable(false);
+        alertDialog.setButton(DialogInterface.BUTTON_POSITIVE, targetActivity.getString(R.string.cancel),
+                (dialog, i) -> {
+                    dialog.dismiss();
+                });
+
+        alertDialog.setButton(DialogInterface.BUTTON_NEGATIVE, targetActivity.getString(R.string.switch_method), listener);
+
+        //the icons set in the dialog are for destinations.
+        if (targetActivity instanceof BtReceiverActivity || targetActivity instanceof BtSenderActivity) {
+            alertDialog.setIcon(R.drawable.ic_wifi_tethering_black_24dp);
+            alertDialog.setTitle(targetActivity.getString(R.string.switch_to_hotspot_title));
+            alertDialog.setMessage(targetActivity.getString(R.string.hotspot_switch_method));
+        } else if (targetActivity instanceof HpReceiverActivity || targetActivity instanceof HpSenderActivity) {
+            alertDialog.setIcon(R.drawable.ic_bluetooth_black_24dp);
+            alertDialog.setTitle(targetActivity.getString(R.string.switch_to_bluetooth_title));
+            alertDialog.setMessage(targetActivity.getString(R.string.bluetooth_switch_method));
+        } else {
+            throw new IllegalArgumentException("Not a valid activity parameter");
+        }
+
+        return alertDialog;
+    }
+
+    /**
+     * Detecting the default set by user, and using that as main sending method.
      */
-    public static AlertDialog showSenderMethodsDialog(Context context, Intent intent, String title) {
-        String[] options = {context.getString(R.string.method_bluetooth), context.getString(R.string.method_wifi_hotspot)};
-        return new AlertDialog.Builder(context)
-                .setTitle(title)
-                .setIcon(R.drawable.ic_help_outline)
-                .setItems(options, (DialogInterface dialog, int which) -> {
-                    if (ActivityUtils.getActivity(context) != null) {
-                        switch (which) {
-                            case 0:
-                                intent.setClass(ActivityUtils.getActivity(context), BtSenderActivity.class);
-                                break;
-                            case 1:
-                                intent.setClass(ActivityUtils.getActivity(context), HpSenderActivity.class);
-                                break;
-                        }
-                        context.startActivity(intent);
-                    }
-                }).create();
+    public static void switchToDefaultSendingMethod(@NonNull Context context, Intent intent) {
```

Comment:
Add `@NonNull` to Intent as well, because for that also we don't have any null check in method then expect it to be nonnull from the caller of this method

Reply:
Sure, will do now.